src/libraries/System.Net.Sockets/src/System/Net/Sockets/TCPClient.cs

```diff
@@ -74,6 +74,9 @@ public TcpClient(string hostname, int port)
                 throw new ArgumentOutOfRangeException(nameof(port));
             }
 
+            _family = AddressFamily.Unknown;
+            InitializeClientSocket();
```

Comment:
Doesn't the following become dead code after this?

https://github.com/dotnet/runtime/blob/c2e098b14a9a352fc9a2afad59060d6097f89418/src/libraries/System.Net.Sockets/src/System/Net/Sockets/TCPClient.cs#L166

Reply:
I did small change so it is not dead again. 
Deferring to Socket.Connect() is luring but I'm not sure if we should do it as part of this fix. The difference is that the Socket.Connect() would depend on dual mode (for all cases) while the current code does not.

Reply:
So the empty constructor + `tcpClient.Connect(host, port)` combo will always use dual mode, while the constructor `TcpClient(host, port)` will trigger a custom connection logic? Why to keep maintaining such a behavioral difference? I think these two cases should work the same way.

Reply:
I was thinking about it and I think we should change it as part of this fix and make unnecessary changes. The constructors fundamentally work different way IMHO. 
You have option to create the TcpClient object either with family or say "I don't care". And then the Connect work accordingly. 
The constructor with host does not allow any customization. Let say you are fully in IPv4 environment. Now the endpoints will be IPv4 addresses. If we switch to dual mode socket, it will become IPv6 mapped address unconditionally. I don't know if that can cause problems but it certainly did in the past. If we want to change the behavior, I think we should do it as separate task and evaluate the risk.